app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,598 @@
+    setTheme(sharedPreferenceUtil.nightMode() ? R.style.Theme_AppCompat_DayNight_NoActionBar : R.style.AppTheme);
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_local_file_transfer);
+    ButterKnife.bind(this);
+
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    Intent filesIntent = getIntent();
+    fileUriArrayList = filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+    if (fileUriArrayList != null && fileUriArrayList.size() > 0) {
+      setDeviceAsFileSender();
+    }
+
+    setSupportActionBar(actionBar);
+    actionBar.setNavigationIcon(R.drawable.ic_close_white_24dp);
+    actionBar.setNavigationOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        wifiDirectManager.closeLocalFileTransferActivity();
+      }
+    });
+
+    wifiDirectManager.initialiseWifiDirectManager();
+
+    listViewPeerDevices.setAdapter(
+        new WifiPeerListAdapter(this, R.layout.row_peer_device, peerDevices));
+
+    if (fileSendingDevice) {
+      totalFilesForTransfer = fileUriArrayList.size();
+
+      for (int i = 0; i < fileUriArrayList.size(); i++) {
+        filesToSend.add(new FileItem(getFileName(fileUriArrayList.get(i)), TO_BE_SENT));
+      }
+
+      displayFileTransferProgress();
+    }
+  }
+
+  @OnItemClick(R.id.list_peer_devices)
+  void onItemClick(int position) {
+    /* Connection can only be initiated by user of the sender device, & only when transfer has not been started */
+    if (!fileSendingDevice || fileTransferStarted) {
+      return;
+    }
+
+    WifiP2pDevice senderSelectedPeerDevice =
+        (WifiP2pDevice) listViewPeerDevices.getAdapter().getItem(position);
+    wifiDirectManager.setSenderSelectedPeerDevice(senderSelectedPeerDevice);
+    alertDialogShower.show(
+        new KiwixDialog.FileTransferConfirmation(senderSelectedPeerDevice.deviceName),
+        new Function0<Unit>() {
+          @Override public Unit invoke() {
+            wifiDirectManager.connect();
+            showToast(LocalFileTransferActivity.this, R.string.performing_handshake,
+                Toast.LENGTH_LONG);
+            return Unit.INSTANCE;
+          }
+        });
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(@NonNull Menu menu) {
+    getMenuInflater().inflate(R.menu.wifi_file_share_items, menu);
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+    if (item.getItemId() == R.id.menu_item_search_devices) {
+
+      /* Permissions essential for this module */
+      if (!checkCoarseLocationAccessPermission()) return true;
```

Comment:
no single line ifs and no ifs without braces

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/314ff811fd402d09bab7d5f2d14cae03a8f8e24d#diff-e1d59ae6e15dcb502e125fb6e243d15dR179)